ipykernel/hookmanager.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+"""
+A context manager that will register a hook on a
+Display Publisher for storing messages.
+"""
+
+# Copyright (c) IPython Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+from io import StringIO
+import sys
+
+from IPython.core.getipython import get_ipython
+
+from .displayhook import ZMQMessageHook
+
+
+class MessageHookFor(object):
+    """
+    A context manager which takes the name of a message field,
+    and installs a hook to intercept messages of that type.
+
+    Usage
+    -----
+    >>> with MessageHookFor('display_data'):
+            clear_output()
+    """
+    def __init__(self, message_name, parent=None):
```

Comment:
Docstring and signature suggest that this is a message for a single type, but it also captures stream messages unconditionally.